### PR TITLE
add/remove students to/from class

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- ticket redmine: https://edu-redmine.sun-asterisk.vn/issues/76123

## WHAT (optional)
- Add/remove students to/from class.

## HOW
- I add interfaces, controllers and some services for this feature. I also fix some UI bug from previous commits.

## Evidence (Screenshot or Video)
image
image
image
image

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/controllers/staff/class.controller.ts [hunk 1]
```diff
@@ -75,6 +75,58 @@ export const getClassesByGrade = async (
...
 
+export const getClassDetail = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const { source } = req.query;
+    const _source = source?.toString() || "";
+    const msg = getSuccessMessage(_source, "student");
+    const success_msg = msg.length > 0 ? req.t(msg) : undefined;
```
Comment (huongnt-2545) [inline src/controllers/staff/class.controller.ts]: ủa tự dưng căn lề vô chi zậy e

File: src/controllers/staff/class.controller.ts [hunk 2]
```diff
@@ -75,6 +75,58 @@ export const getClassesByGrade = async (
...
+  const id = parseInt(req.params.id);
```
Comment (huongnt-2545) [inline src/controllers/staff/class.controller.ts]:
```suggestion
  const id = +req.params.id;
```

File: src/controllers/staff/class.controller.ts [hunk 3]
```diff
@@ -75,6 +75,58 @@ export const getClassesByGrade = async (
...
+};
+
+export const addStudentToClass = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const classId = parseInt(req.params.id);
+  const students = req.body.students;
+  const studentIds = students.map(Number);
```
Comment (huongnt-2545) [inline src/controllers/staff/class.controller.ts]: Thêm xử lý check nếu student ko tồn tại

File: src/controllers/staff/class.controller.ts [hunk 4]
```diff
@@ -75,6 +75,58 @@ export const getClassesByGrade = async (
...
+export const removeStudentFromClass = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const classId = parseInt(req.params.id);
+  const students = req.body.students;
+  const studentIds = students.map(Number);
+  if (studentIds.includes(NaN))
+    return res.json({ errors: { student: "class.student_invalid" } });
```
Comment (huongnt-2545) [inline src/controllers/staff/class.controller.ts]: phần xử lý này move ra function khác để dùng chung e nha

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]

**Pull request merged**